### ISSUE TITLE
"[WARNING] unknown enum constant org.osgi.annotation.bundle.Requirement.Resolution.OPTIONAL" when compiler warnings are enabled

### ISSUE THREAD

**Issue by famod** [quarkusio/quarkus#19970]
### Describe the bug

`maven-compiler-plugin` issues a warning when your app is using `@ConfigProperty` and compiler warnings are enabled:
```
[INFO] Compiling 1 source file to /home/famod/proj/quarkus-quickstarts/config-quickstart/target/classes
[WARNING] unknown enum constant org.osgi.annotation.bundle.Requirement.Resolution.OPTIONAL
  reason: class file for org.osgi.annotation.bundle.Requirement$Resolution not found
```

### Expected behavior

No such warning.

### Actual behavior

Warning w.r.t OSGI annotations. Also pollutes "real" compilation errors which can lead developers astray.

### How to Reproduce?

`./mvnw clean compile -f config-quickstart/ -Dmaven.compiler.showWarnings=true` in `quarkus-quickstarts`.

### Output of `uname -a` or `ver`

Linux XXX 5.10.16.3-microsoft-standard-WSL2 #1 SMP Fri Apr 2 22:23:49 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

OpenJDK 64-Bit Server VM Temurin-11.0.12+7 (build 11.0.12+7, mixed mode)

### Quarkus version or git rev

2.1.2.Final, probably since 2.0, up to current 999-SNAPSHOT

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.1

### Additional information

https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/OSGI.20Warnings

https://mvnrepository.com/artifact/org.eclipse.microprofile.config/microprofile-config-api/2.0 defines provided OSGI annotations.
Adding https://mvnrepository.com/artifact/org.osgi/osgi.annotation/7.0.0 works around the issue.

**Comment by radcortez** [comment]
Unfortunately, there is nothing we can do on the Quarkus side because this is caused by an issue in the MP Config API. Opened https://github.com/eclipse/microprofile-config/issues/716 to see if we can fix it there.

**Comment by famod** [comment]
Thanks for looking into it!
I expected this to be an upstream issue right from the start and not something that can be fixed in SR Config (or even Quarkus).
That's why I added the upstream label and so I'm now a little surprised that you removed it. 🤔
But anyway, a label is just a label. 😉

**Comment by radcortez** [comment]
Sorry, I've removed it because I thought this is already "triaged" and that we now have a linked issue to the upstream dependency.

**Comment by famod** [comment]
Ah ok. I thought we also use it to flag issues that are waiting for upstream changes. 🤷‍♂️

**Comment by bjhargrave** [comment]
FYI, I am working a fix to the OSGi annotations to replace use of enum types with String along with a corresponding fix to Bnd to support this change.

**Comment by gsmet** [comment]
I confirm the `upstream` label is added to mark issues that are waiting for an upstream fix. Ultimately, I'd like to remove them from our stats as there's nothing we can do about them.

**Comment by bjhargrave** [comment]
FYI, https://github.com/eclipse/microprofile-config/pull/732 has been filed to update MicroProfile Config API to avoid the compiler warning issue.

**Comment by famod** [comment]
@MarcusBiel 2.7.3 should have fixed this (I haven't checked yet).

**Comment by gastaldi** [comment]
I can confirm this is fixed in 2.7.3.Final

**Comment by MarcusBiel** [comment]
Can't confirm since 2.7.3 doesn't work for me:
Build failure: Build failed due to errors
        [error]: Build step io.quarkiverse.logging.logback.deployment.LoggingLogbackProcessor#init threw an exception: java.lang.NoClassDefFoundError: io/quarkus/bootstrap/resolver/update/DefaultArtifactVersion
        at io.quarkiverse.logging.logback.deployment.LoggingLogbackProcessor.doVersionCheck(LoggingLogbackProcessor.java:196)
        at io.quarkiverse.logging.logback.deployment.LoggingLogbackProcessor.init(LoggingLogbackProcessor.java:92)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)

--
Probably a regression with https://quarkiverse.github.io/quarkiverse-docs/quarkus-logging-logback/dev/index.html
?

**Comment by gastaldi** [comment]
Better open an issue in the respective quarkiverse extension with a small reproducer

**Comment by famod** [comment]
> I can confirm this is fixed in 2.7.3.Final

Me too now.
